src/python/pants/engine/scheduler.py

```diff
@@ -559,14 +560,15 @@ def products_request(self, products, subjects):
       product_results[product].append(state.value)
     return product_results
 
-  def product_request(self, product, subjects):
+  def product_request(self, product, subjects, render_v2_engine_ui=False):
```

Comment:
As discussed on #6654, I think that you should add a `@console_rule` specific method here, and remove the TODO that Daniel added: https://github.com/pantsbuild/pants/blob/3d22e955e6a4a124669941c867f1598d81787f86/src/python/pants/engine/scheduler.py#L541-L542

The idea would be that: a call to `product_request(s)` would never render the UI, but a call to this new method would, and would also address that TODO. It would also not support batching like `product_requests` (plural) does.

Reply:
Made a new method specifically for console rules. However it is not clear to me whether the entire `if` statement was specifically for console rules, thus can be removed?
https://github.com/pantsbuild/pants/blob/1b77e5645ab1caa1f317d714bef2fe4e4f4c3968/src/python/pants/engine/scheduler.py#L591-L613